### PR TITLE
 check kernel for auto freeze

### PR DESCRIPTION
**Pull request by Qi77Qi**
Questions:
1. What's the best way to test out the jupyter call?

Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get a thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/HttpJupyterDAO.scala [hunk 1]
```diff
@@ -32,4 +38,41 @@ class HttpJupyterDAO(val clusterDnsCache: ClusterDnsCache)(implicit system: Acto
     implicit val timeout: Timeout = Timeout(5 seconds)
     clusterDnsCache.getHostStatus(DnsCacheKey(googleProject, clusterName)).mapTo[HostStatus]
   }
+
+  override def isAllKernalsIdle(googleProject: GoogleProject, clusterName: ClusterName): Future[Boolean] = {
+    for {
+      hostStatus <- getTargetHost(googleProject, clusterName)
+      resp <- hostStatus match {
+        case HostReady(host) =>
+          val sessionUri = Uri(s"https://${host.toString}/notebooks/$googleProject/$clusterName/api/sessions") //TODO: validate this url is correct
```
Comment (vkumra-broad) [inline src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/HttpJupyterDAO.scala]: Url looks good.

File: src/test/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ClusterMonitorSupervisorSpec.scala [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+      clusterDefaultsConfig, proxyConfig, swaggerConfig, autoFreezeConfig, gdDAO, computeDAO, iamDAO,
+      storageDAO, mockPetGoogleStorageDAO, DbSingleton.ref, whitelistAuthProvider, serviceAccountProvider, whitelist,
+      bucketHelper, contentSecurityPolicy)
+
+    val clusterSupervisorActor = system.actorOf(ClusterMonitorSupervisor.props(monitorConfig, dataprocConfig, clusterBucketConfig, gdDAO,
+      computeDAO, iamDAO, storageDAO, DbSingleton.ref, authProvider, autoFreezeConfig, jupyterProxyDAO, MockRStudioDAO, leoService))
+
+    eventually(timeout(Span(30, Seconds))) {
+      val c1 = dbFutureValue { _.clusterQuery.getClusterById(runningCluster.id) }
+      c1.map(_.status) shouldNot be (Some(ClusterStatus.Stopped))
```
Comment (Qi77Qi) [inline src/test/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ClusterMonitorSupervisorSpec.scala]: not sure exactly if this is valid check. Would be nice if I can check `c1.map(_.status) shouldBe (Some(ClusterStatus.Running))`. But seems like cluster is not necessarily running  at this point.

**Comment by vkumra-broad** [comment]
LGTM if automation tests pass.

**Comment by Qi77Qi** [comment]
```
--2019-04-24 18:54:16--  https://download.oracle.com/otn-pub/java/jdk/8u201-b09/42970487e3af4f5aa5bca3f542482c60/jdk-8u201-linux-x64.tar.gz
Resolving download.oracle.com (download.oracle.com)... 23.45.132.164
Connecting to download.oracle.com (download.oracle.com)|23.45.132.164|:443... connected.
HTTP request sent, awaiting response... 302 Moved Temporarily
```

jenkins retest

**Comment by Qi77Qi** [comment]
jenkins retest

just to check if it's temparary

**Comment by vkumra-broad** [comment]
It might be useful to add a log message along the lines of -- 
```
cluster not auto paused because of active kernel session
```
if there is an active kernel session

Thoughts?

**Pull request merged**